ssc/cmod_grid.cpp

```diff
@@ -128,20 +128,22 @@ void cm_grid::exec() throw (general_error)
         load_year_one = as_vector_double("load");
     }
 
+
+    /* Does nothing since "load_escalation" not an input to this compute module
     scalefactors scale_calculator(m_vartab);
 
     // compute load (electric demand) annual escalation multipliers
     std::vector<ssc_number_t> load_scale;
     if (is_assigned("load_escalation")) {
         load_scale = scale_calculator.get_factors("load_escalation");
     }
-
+    */
     single_year_to_lifetime_interpolated<double>(
         system_use_lifetime_output,
         analysis_period,
         n_rec_lifetime,
         load_year_one,
-        load_scale,
+        scaleFactors,
```

Comment:
The escalation of the load in the pv and battery models was added for proper lifetime battery dispatch signals-- as the load goes up over time, the battery must also try to meet that load to capture the financial benefits. This would result in higher `gen` output over the lifetime since `gen` is prior to `load` subtraction. This is now applied for both BTM and FOM.

For proper financial calculations where `cmod_grid` is included, the interconnect and curtailment limitations are applied to `gridNet = gen - gridVars->loadLifetime_kW[i]` in line 162/4. I think both `gen` and `loadLifetime_kW` were not escalated. 

But now if it were a battery simulation, `gen` is escalated since it was dispatched with escalated loads, so this scaleFactor should not just be `std::vector<double> scaleFactors(analysis_period, 1.0);`. However, if it were not a battery simulation, then `gen` will not be escalated and neither should the load. This means the value of scaleFactors should depend on whether or not `load_escalation` is available and makes sense.... Or am I missing something here?

Reply:
The escalated load parameters apply to all pvsamv1, cmod battery, battwatts, and solar water heater simulations, as well as the utility rate modules. So I think you're correct that `load_scale` should apply here when we're doing lifetime simulations. `gen` should already be lifetime length, and it would account for PV degradation in most cases.

The question then, is what to do when the analysis period is 1 year (and, as a result, is undefined), or we're doing AC side degradation in the financial models (like for PVWatts)? It's worth noting that the GUI has load escalation defined, so it might be worth requiring some additional variables in the vartable/in the SAM defaults.